### PR TITLE
Fix variation popover custom attribute matching

### LINKED ISSUE #398: POS variations popover disables options when custom attributes share id 0
## Summary

A client reported that the POS product variations popover disables the second row of options for a variable product whose variation attributes are both custom/local attributes with `id: 0`.

Example parent product:

- Product: `Antifreeze` (`id: 26932`)
- Attributes:
  - `Colour` (`id: 0`): `Blue`, `Red`, `Yellow`
  - `Size` (`id: 0`): `1L`, `5L`, `25L`
- Variations: 9 total, covering the colour/size matrix

## Current behavior

When opening the variations popover in the POS Cart/Product table, the bottom row of options is inactive/disabled:

```text
1L | 5L | 25L
```

## Expected behavior

The `Size` options should be active when matching variations exist.

## Investigation notes

Preliminary root cause appears to be in:

```text
packages/core/src/screens/main/pos/products/cells/variations-popover/utils.ts
```

`hitMatchesOption()` identifies variation attributes by `id` only:

```ts
const targetHitAttr = hitAttrs.find((a) => a.id === targetAttrId);
```

WooCommerce custom/local attributes commonly use `id: 0`, so both `Colour` and `Size` have the same id. When checking the `Size` row, the matcher can pick the `Colour` attribute first and compare values like `Blue !== 1L`, causing every size option count to become `0` and the UI disables those buttons.

The matching logic should use the logical attribute identity (`id` + `name`, or `name` when `id === 0`) instead of `id` alone.

## Acceptance criteria

- Add a regression test for a variable product with multiple custom attributes where each parent and variation attribute has `id: 0`.
- The variation popover option counts should remain positive for valid options in later rows, e.g. `1L`, `5L`, `25L`.
- Existing “any option” variation behavior should continue to pass.

### PR DESCRIPTION
**Pull request by kilbot**
## Summary

- Fixes the POS variations popover option-count logic for WooCommerce custom/local attributes where multiple attributes share `id: 0`.
- Matches variation attributes by both `id` and `name` so rows like `Colour` and `Size` no longer collapse into the first id=0 attribute.
- Adds a regression test covering the reported Antifreeze-style matrix (`Colour` x `Size`) where all variation attributes have `id: 0`.

Fixes #398.

## Test plan

- [x] `pnpm --filter @wcpos/core test -- src/screens/main/pos/products/cells/variations-popover/utils.test.ts --runInBand`
- [x] `pnpm --filter @wcpos/core lint` (passes with 16 pre-existing warnings, 0 errors)
- [x] `pnpm --filter @wcpos/core typecheck`
- [ ] In POS Products/Cart table view, open the variations popover for a variable product with custom attributes `Colour` and `Size` both using `id: 0`; confirm `1L`, `5L`, and `25L` are active.
- [ ] Select a valid colour/size combination and confirm the Add to Cart button appears for the resolved variation.
- [ ] Smoke-test an “Any option” variation product to confirm existing wildcard behavior still works.

## Review notes

- `codex review --base main` reported no discrete correctness issue introduced by this diff.
- The `/critic` subagent gate was skipped because this Codex environment only permits spawning subagents when explicitly requested by the user.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed variations popover so custom/local attributes (using id 0) correctly show all available options and no options are incorrectly preselected; matching now considers attribute identity by name when needed.

* **Tests**
  * Added tests covering id-0/custom attribute edge cases and full attribute-combination coverage to prevent regressions.

### CODE REVIEW

File: packages/core/src/screens/main/pos/products/cells/variations-popover/utils.ts [hunk 1]
```diff
@@ -36,6 +36,18 @@ import type { ProductDocument, ProductVariationDocument } from '@wcpos/database'
...
+/**
+ * WooCommerce global attributes have stable IDs, but custom/local attributes use id=0.
+ * Match on both id and name so multiple custom attributes do not collapse together.
+ */
+function isSameAttribute(
+	attribute: { id?: number; name?: string },
+	targetAttrId: number,
+	targetAttrName: string
+): boolean {
+	return attribute.id === targetAttrId && attribute.name === targetAttrName;
```
[the remark this answers is not in the document]
Comment (kilbot) [reply packages/core/src/screens/main/pos/products/cells/variations-popover/utils.ts]: Fixed in dc0fcaa8. Nonzero attributes now preserve the id match when variation attribute names are absent, while still checking names when both sides provide them.

File: packages/core/src/screens/main/pos/products/cells/variations-popover/utils.ts [hunk 2]
```diff
@@ -36,6 +36,18 @@ import type { ProductDocument, ProductVariationDocument } from '@wcpos/database'
...
+}
```
[the remark this answers is not in the document]
Comment (kilbot) [reply packages/core/src/screens/main/pos/products/cells/variations-popover/utils.ts:57]: Fixed in dc0fcaa8. Custom/local attributes with target id 0 now match by name alone, so serialized variation attributes without an id are handled correctly.
